lm_eval/tasks/__init__.py

```diff
@@ -476,8 +510,16 @@ def get_task_dict(
     ):
         raise ValueError
 
-    return {
+    final_task_dict = {
         **task_name_from_string_dict,
         **task_name_from_config_dict,
         **task_name_from_object_dict,
     }
+
+    # behavior can get odd if one tries to invoke several groups that "compete" for the same task.
+    # (notably, because one could request several num_fewshot values at once in GroupConfig overrides for the subtask
+    # and we'd be unsure which to use and report.)
+    # we explicitly check and error in this case.
+    _check_duplicates(get_subtask_list(final_task_dict))
+
+    return final_task_dict
```

Comment:
Just to make sure, so our position should be to not tolerate duplicate tasks (same task name, same task config) even if they are in a different group?

Reply:
That was my thought for now--and then we can down the line maybe do something like: 

after updating w/ num_fewshot, etc, 2 subtasks in overlapping groups have *the exact same config* then we can be okay with that?

Reply:
The main issue is that there can be duplicate task names in different groups. If there are 2 or more tasks with the same name under a single group (say we run `boolq` but with num_fewshot of 2,4,and 8) they will be added additional numbering. I guess it would be okay to have lm-eval exit for now.